core/src/main/java/org/frankframework/errormessageformatters/ErrorMessageFormatter.java

```diff
@@ -86,42 +94,75 @@ public Message format(String errorMessage, Throwable t, HasName location, Messag
 		errorMessage = StringUtil.concatStrings(prefix, ": ", errorMessage);
 
 		String originator = AppConstants.getInstance().getProperty("application.name")+" "+ AppConstants.getInstance().getProperty("application.version");
-		// Build a Base xml
-		XmlBuilder errorXml = new XmlBuilder("errorMessage");
-		errorXml.addAttribute("timestamp", new Date().toString());
-		errorXml.addAttribute("originator", originator);
-		errorXml.addAttribute("message", XmlEncodingUtils.replaceNonValidXmlCharacters(errorMessage));
-
-		if (location != null) {
-			XmlBuilder locationXml = new XmlBuilder("location");
-			locationXml.addAttribute("class", location.getClass().getName());
-			locationXml.addAttribute("name", location.getName());
-			errorXml.addSubElement(locationXml);
-		}
-
-		if (StringUtils.isNotEmpty(details)) {
-			XmlBuilder detailsXml = new XmlBuilder("details");
-			// detailsXml.setCdataValue(details);
-			detailsXml.setValue(XmlEncodingUtils.replaceNonValidXmlCharacters(details), true);
-			errorXml.addSubElement(detailsXml);
+		// Build a Base document
+		try {
+			MessageBuilder messageBuilder = new MessageBuilder();
+			IDocumentBuilder documentBuilder = DocumentBuilderFactory.startDocument(messageFormat, "errorMessage", messageBuilder, true);
+
+			ObjectBuilder errorObject;
+			ObjectBuilder rootObjectBuilder;
+			if (messageFormat == DocumentFormat.XML) {
+				rootObjectBuilder = null;
+				errorObject = documentBuilder.asObjectBuilder();
+			} else {
+				rootObjectBuilder = documentBuilder.asObjectBuilder();
+				errorObject = rootObjectBuilder.addObjectField("errorMessage");
+			}
+			errorObject.addAttribute("timestamp", new Date().toString());
+			errorObject.addAttribute("originator", originator);
+			errorObject.addAttribute("message", XmlEncodingUtils.replaceNonValidXmlCharacters(errorMessage));
```

Comment:
In de `SaxElementBuilder#addAttribute(String, String)` zit ook al een replace xml ding dacht ik. Misschien kan deze wel weg :)